nucliadb_utils/src/nucliadb_utils/aiopynecone/models.py

```diff
@@ -20,14 +20,47 @@
 import json
 from typing import Any, Optional
 
+import pydantic
 from pydantic import BaseModel, Field, field_validator
+from typing_extensions import Annotated
 
 KILO_BYTE = 1024
 MAX_METADATA_SIZE = 40 * KILO_BYTE
+MAX_INDEX_NAME_LENGTH = 45
 
 
 # Requests
 
+IndexNamePattern = r"^[a-z0-9-]+$"
+
+
+def validate_index_name(value, handler, info):
+    try:
+        return handler(value)
+    except pydantic.ValidationError as e:
+        if any(x["type"] == "string_pattern_mismatch" for x in e.errors()):
+            raise ValueError(
+                f"Invalid field_id: '{value}'. Pinecone index names must be a string with only "
+                "lowercase letters, numbers and dashes."
+            )
+        else:
+            raise e
+
+
+IndexNameStr = Annotated[
+    str,
+    pydantic.StringConstraints(pattern=IndexNamePattern),
+    pydantic.StringConstraints(min_length=1, max_length=MAX_INDEX_NAME_LENGTH),
+    pydantic.WrapValidator(validate_index_name),
```

Comment:
Couldn't this be written as:
```suggestion
    pydantic.StringConstraints(
        min_length=1, 
        max_lenght=MAX_INDEX_NAME_LENGTH,
        pattern=IndexNamePattern
    ),
```

Reply:
will change in a future pr